lib/segment/src/index/field_index/full_text_index/text_index.rs

```diff
@@ -97,22 +111,19 @@ impl ValueIndexer<String> for FullTextIndex {
             return Ok(());
         }
 
-        let mut tokens: HashSet<String> = HashSet::new();
+        let mut tokens: BTreeSet<String> = BTreeSet::new();
 
         for value in values {
             Tokenizer::tokenize_doc(&value, &self.config, |token| {
                 tokens.insert(token.to_owned());
             });
         }
 
-        let document = Document {
-            tokens: tokens.into_iter().collect(),
-        };
-
+        let document = self.inverted_index.document_from_tokens(tokens);
         self.inverted_index.index_document(idx, document);
 
         let db_idx = Self::store_key(&idx);
-        let db_document = Self::serialize_document(
+        let db_document = self.serialize_document(
```

Comment:
we already have _tokens_ here, we don't need to revert vocabulary here, just serialize tokens as is